example/client/include/KVMessageGenerator.hpp

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <chrono>
+#include <map>
+#include <set>
+#include <vector>
+#include <cstdio>
+#include <cstring>
+#include <sys/param.h>
+#include <unistd.h>
+#include <yaml-cpp/yaml.h>
+#include <TimeUtils.hpp>
+
+#include "Logger.hpp"
+#include "kv_replica_msgs.cmf.hpp"
+#include "messages/ClientRequestMsg.hpp"
+#include "KVBCInterfaces.h"
+#include "common.hpp"
+#include "block_metadata.hpp"
+#include "boost/detail/endian.hpp"
+#include "assertUtils.hpp"
+#include "storage/db_types.h"
+#include "IMessageGenerator.hpp"
+
+namespace concord::kvbc {
+using namespace ::kvbc::messages;
+
+struct KVBlock {
+  uint64_t id = 0;
+  std::vector<std::pair<std::vector<uint8_t>, std::vector<uint8_t>>> items;
+};
+typedef std::map<std::pair<std::vector<uint8_t>, uint64_t>, std::vector<uint8_t>> KeyBlockIdToValueMap;
+
+class KVMessageGenerator : public IMessageGenerator {
+ public:
+  KVMessageGenerator(SampleMessage* sampleMsg) : sampleMsg_(sampleMsg) {}
+  ~KVMessageGenerator() = default;
+  void setMessageFileName(const std::string& fileName) { sampleMsgFileName_ = fileName; }
+
+  bft::client::Msg createWriteRequestMessage() override;
+  bft::client::Msg createReadRequestMessage() override;
+
+ private:
+  logging::Logger getLogger() {
+    static logging::Logger logger_(logging::getLogger("osexample::KVMessageGenerator"));
+    return logger_;
+  }
+
+  void addExpectedWriteReply(bool foundConflict);
+  void addExpectedReadReply(size_t numberOfKeysToRead, KVReadRequest& read_req);
+  bool lookForConflicts(uint64_t readVersion, const std::vector<std::vector<uint8_t>>& readKeysArray);
+  void addNewBlock(const std::vector<std::pair<std::vector<uint8_t>, std::vector<uint8_t>>>& writesKVArray);
+
+ private:
+  std::string sampleMsgFileName_;
+  SampleMessage* sampleMsg_;
```

Comment:
can we use unique_ptr or shared_ptr here?

Reply:
fixed